lib/modules/blockchain_process/proxy.js

```diff
@@ -127,6 +146,9 @@ exports.serve = function (ipc, host, port, ws) {
     });
   }
   const listenPort = port - constants.blockchain.servicePortOnProxy;
-  server.listen(listenPort, defaultHost);
-  return server;
+  return new Promise(resolve => {
+    server.listen(listenPort, defaultHost, () => {
+      resolve(server);
```

Comment:
the server is returned only when started

Reply:
Really slick!

Reply:
Awesome!